File: bnim2219-spring/src/main/java/edu/bbte/idde/bnim2219/spring/model/Chore.java
package edu.bbte.idde.bnim2219.spring.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import lombok.*;

import java.sql.Date;

@Data
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
@ToString(callSuper = true)
@Entity
public class Chore extends BaseEntity {
    @Column(nullable = false, length = 50)
    private String title;
    private String description;
    @Column(nullable = false)
    private Date deadline;
    @Column(nullable = false)
    private Integer priorityLevel;
    @Column(nullable = false)
    private Boolean done;
}

File: src/model/chore.py
from sqlalchemy.orm import relationship

from src.model.base_entity import BaseEntity
from src.db import db


class Chore(BaseEntity):
    title = db.Column(db.String(50), nullable=False)
    description = db.Column(db.String(255))
    deadline = db.Column(db.Date, nullable=False)
    priorityLevel = db.Column(db.Integer, nullable=False)
    done = db.Column(db.Boolean, default=False, nullable=False)
    subtasks = relationship('Subtask', back_populates='chore', cascade='all, delete-orphan')
